File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/exporter/options/ExportOptions.java
/*
 * Copyright ETH 2023 Zürich, Scientific IT Services
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.exporter.options;

import java.io.Serializable;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.exporter.options.ExportOptions")
public class ExportOptions implements Serializable
{

    private static final long serialVersionUID = 1L;

    @JsonProperty
    private Set<ExportFormat> formats;

    @JsonProperty
    private XlsTextFormat xlsTextFormat;

    @JsonProperty
    private Boolean withReferredTypes;

    @JsonProperty
    private Boolean withImportCompatibility;

    @JsonProperty
    private Boolean zipSingleFiles;

    @SuppressWarnings("unused")
    public ExportOptions()
    {
    }

    public ExportOptions(final Set<ExportFormat> formats, final XlsTextFormat xlsTextFormat, final Boolean withReferredTypes,
            final Boolean withImportCompatibility, final Boolean zipSingleFiles)
    {
        this.formats = formats;
        this.xlsTextFormat = xlsTextFormat;
        this.withReferredTypes = withReferredTypes;
        this.withImportCompatibility = withImportCompatibility;
        this.zipSingleFiles = zipSingleFiles;
    }

    @JsonIgnore
    public Set<ExportFormat> getFormats()
    {
        return formats;
    }

    @JsonIgnore
    public void setFormats(final Set<ExportFormat> formats)
    {
        this.formats = formats;
    }

    @JsonIgnore
    public XlsTextFormat getXlsTextFormat()
    {
        return xlsTextFormat;
    }

    @JsonIgnore
    public void setXlsTextFormat(final XlsTextFormat xlsTextFormat)
    {
        this.xlsTextFormat = xlsTextFormat;
    }

    @JsonIgnore
    public Boolean isWithReferredTypes()
    {
        return withReferredTypes;
    }

    @JsonIgnore
    public void setWithReferredTypes(final Boolean withReferredTypes)
    {
        this.withReferredTypes = withReferredTypes;
    }

    @JsonIgnore
    public Boolean isWithImportCompatibility()
    {
        return withImportCompatibility;
    }

    @JsonIgnore
    public void setWithImportCompatibility(final Boolean withImportCompatibility)
    {
        this.withImportCompatibility = withImportCompatibility;
    }

    @JsonIgnore
    public Boolean isZipForSingleFiles()
    {
        return zipSingleFiles;
    }

    @JsonIgnore
    public void setZipSingleFiles(final Boolean zipSingleFiles)
    {
        this.zipSingleFiles = zipSingleFiles;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("formats", formats).append("xlsTextFormat", xlsTextFormat)
                .append("withReferredTypes", withReferredTypes).append("withImportCompatibility", withImportCompatibility).toString();
    }

}

File: api-openbis-javascript/src/v3/as/dto/exporter/options/ExportOptions.js
/*
 *  Copyright ETH 2023 Zürich, Scientific IT Services
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

define(["stjs"], function (stjs) {
  var ExportOptions = function() {
  }

  stjs.extend(
    ExportOptions,
    null,
    [],
    function (constructor, prototype) {
      prototype["@type"] = "as.dto.exporter.options.ExportOptions";

      constructor.serialVersionUID = 1;
      prototype.formats = null;
      prototype.xlsTextFormat = null;
      prototype.withReferredTypes = null;
      prototype.withImportCompatibility = null;

      prototype.getFormats = function() {
        return this.formats;
      };

      prototype.setFormats = function(formats) {
        this.formats = formats;
      };

      prototype.getXlsTextFormat = function() {
        return this.xlsTextFormat;
      };

      prototype.setXlsTextFormat = function(xlsTextFormat) {
        this.xlsTextFormat = xlsTextFormat;
      };

      prototype.isWithReferredTypes = function() {
        return this.withReferredTypes;
      };

      prototype.setWithReferredTypes = function(withReferredTypes) {
        this.withReferredTypes = withReferredTypes;
      };

      prototype.isWithImportCompatibility = function() {
        return this.withImportCompatibility;
      };

      prototype.setWithImportCompatibility = function(withImportCompatibility) {
        this.withImportCompatibility = withImportCompatibility;
      };
    },
    {
      formats: {
        name: "Set",
        arguments: ["ExportFormat"]
      },
      xlsTextFormat: "XlsTextFormat",
      withReferredTypes: "Boolean",
      withImportCompatibility: "Boolean"
    }
  );

  return ExportOptions;
});